sys/include/net/nanocoap.h

```diff
@@ -182,13 +181,24 @@ extern "C" {
  * @name    Content-Format option codes
  * @{
  */
-#define COAP_FORMAT_TEXT         (0)
-#define COAP_FORMAT_LINK        (40)
-#define COAP_FORMAT_OCTET       (42)
-#define COAP_FORMAT_JSON        (50)
-#define COAP_FORMAT_CBOR        (60)
-/** @brief   nanocoap-specific value to indicate no format specified. */
-#define COAP_FORMAT_NONE     (65535)
+#define COAP_FORMAT_TEXT            (0)
+#define COAP_FORMAT_LINK           (40)
+#define COAP_FORMAT_OCTET          (42)
+#define COAP_FORMAT_JSON           (50)
+#define COAP_FORMAT_CBOR           (60)
+/**
+ * @brief    nanocoap-internal value to indicate no format specified.
+ *
+ * Not intended for use in a transmitted packet.
+ */
+#define COAP_FORMAT_NONE        (65535)
+/**
+ * @brief    nanocoap-internal value to indicate no payload present.
+ *
+ * Not intended for use in a transmitted packet. Must be a 3-byte unsigned
+ * value.
+ */
+#define COAP_FORMAT_NO_PAYLOAD  (65536)
```

Comment:
and just noticed: It seems to me this define is only used in gcoap.c. So I think it makes sense to make it local to that file, right?!

Reply:
Absolutely, it's a gcoap internal value.